src/runtime/recipe/handle-connection.ts

```diff
@@ -209,27 +212,27 @@ export class HandleConnection {
     return true;
   }
 
-  _resetHandleType() {
+  private _resetHandleType(): void {
     if (this._handle) {
       this._handle._type = undefined;
     }
   }
 
-  connectToHandle(handle) {
+  connectToHandle(handle: Handle): void {
     assert(handle.recipe === this.recipe);
     this._handle = handle;
     this._resetHandleType();
     this._handle.connections.push(this);
   }
 
-  disconnectHandle() {
+  disconnectHandle(): void {
     const idx = this._handle.connections.indexOf(this);
     assert(idx >= 0);
     this._handle.connections.splice(idx, 1);
     this._handle = undefined;
   }
 
-  toString(nameMap, options) {
+  toString(nameMap, options): string {
     const result = [];
     result.push(this.name || '*');
     // TODO: better deal with unspecified direction.
```

Comment:
Line 259: `findSpecsForUnnamedHandles` should have the same return type as `spec.handleConnections`

Reply:
next PR :)